apps/pipelines/models.py

```diff
@@ -27,6 +28,13 @@
 from apps.utils.models import BaseModel
 
 
+class ModelParamSpec(TypedDict):
```

Comment:
Making this a dataclass would allow attribute access instead of having to use item getters:

`spec.param_name` instead of `spec["param_name"]`

Reply:
https://github.com/dimagi/open-chat-studio/pull/1405/commits/889296f1be686ac56c2889c7a6d20a02cce7d33e